code/modules/atmospherics/gasmixtures/zextools_broke.dm

```diff
@@ -0,0 +1,304 @@
+
+	removed.set_temperature(temperature)
+	for(var/id in cached_gases)
+		removed_gases[id] = QUANTIZE(cached_gases[id] * ratio)
+		cached_gases[id] -= removed_gases[id]
+
+	GAS_GARBAGE_COLLECT(gases)
+
+	return removed
+
+/datum/gas_mixture/copy()
+	var/list/cached_gases = gases
+	var/datum/gas_mixture/copy = new type
+	var/list/copy_gases = copy.gases
+
+	copy.set_temperature(temperature)
+	for(var/id in cached_gases)
+		copy_gases[id] = cached_gases[id]
+
+	return copy
+
+
+/datum/gas_mixture/copy_from(datum/gas_mixture/sample)
+	var/list/cached_gases = gases //accessing datum vars is slower than proc vars
+	var/list/sample_gases = sample.gases
+
+	temperature = sample.temperature
+	for(var/id in sample_gases)
+		cached_gases[id] = sample_gases[id]
+
+	//remove all gases not in the sample
+	cached_gases &= sample_gases
+
+	return 1
+
+/datum/gas_mixture/copy_from_turf(turf/model)
+	parse_gas_string(model.initial_gas_mix)
+
+	//acounts for changes in temperature
+	var/turf/model_parent = model.parent_type
+	if(model.temperature != initial(model.temperature) || model.temperature != initial(model_parent.temperature))
+		temperature = model.temperature
+
+	return 1
+
+/datum/gas_mixture/share(datum/gas_mixture/sharer, atmos_adjacent_turfs = 4)
+
+	var/list/cached_gases = gases
+	var/list/sharer_gases = sharer.gases
+
+	var/temperature_delta = temperature_archived - sharer.temperature_archived
+	var/abs_temperature_delta = abs(temperature_delta)
+
+	var/old_self_heat_capacity = 0
+	var/old_sharer_heat_capacity = 0
+	if(abs_temperature_delta > MINIMUM_TEMPERATURE_DELTA_TO_CONSIDER)
+		old_self_heat_capacity = heat_capacity()
+		old_sharer_heat_capacity = sharer.heat_capacity()
+
+	var/heat_capacity_self_to_sharer = 0 //heat capacity of the moles transferred from us to the sharer
+	var/heat_capacity_sharer_to_self = 0 //heat capacity of the moles transferred from the sharer to us
+
+	var/moved_moles = 0
+	var/abs_moved_moles = 0
+
+	//we're gonna define these vars outside of this for loop because as it turns out, var declaration is pricy
+	var/delta
+	var/gas_heat_capacity
+	//and also cache this shit rq because that results in sanic speed for reasons byond explanation
+	var/list/cached_gasheats = GLOB.meta_gas_specific_heats
+	//GAS TRANSFER
+	for(var/id in cached_gases | sharer_gases) // transfer gases
+
+		delta = QUANTIZE(gas_archive[id] - sharer.gas_archive[id])/(atmos_adjacent_turfs+1) //the amount of gas that gets moved between the mixtures
+
+		if(delta && abs_temperature_delta > MINIMUM_TEMPERATURE_DELTA_TO_CONSIDER)
+			gas_heat_capacity = delta * cached_gasheats[id]
+			if(delta > 0)
+				heat_capacity_self_to_sharer += gas_heat_capacity
+			else
+				heat_capacity_sharer_to_self -= gas_heat_capacity //subtract here instead of adding the absolute value because we know that delta is negative.
+
+		cached_gases[id]					-= delta
+		sharer_gases[id]			+= delta
+		moved_moles			+= delta
+		abs_moved_moles		+= abs(delta)
```

Comment:
This looks wrong, are the indents a github thing? also wholesome share() living through everything